src/reducers/filter.js

```diff
@@ -0,0 +1,12 @@
+const CHANGE_FILTER = 'CHANGE_FILTER';
+
+const filterReducer = (state = 'All', action) => {
+  switch (action.type) {
+    case CHANGE_FILTER:
+      return [...state, action.book];
+    default:
+      return state;
+  }
+};
```

Comment:
- [ ] Since the default state is a string, a dispatched action shouldn't change it to an array. Simply return `action.id` rather than an array:
```
case CHANGE_FILTER:
  return action.id;
```
- See comment below.